examples/experimental/imagenet_nas_arch_pytorch/model_def.py

```diff
@@ -208,3 +316,26 @@ def evaluate_batch(self, batch: Any, model: nn.Module) -> Dict[str, Any]:
         top1, top5 = accuracy(logits, target, topk=(1, 5))
 
         return {"loss": loss, "top1_accuracy": top1, "top5_accuracy": top5}
+
+    # This function is never called by PyTorch Trial but can be used manually
+    # to evaluate model on full validation dataset.
+    def evaluate_epoch(
```

Comment:
Question: little confused about why we are adding this? 1. how does this differ from evaluate_batch? If we want to use a full validation dataset, we should just use evaluate_full_dataset(). However, part of the perks for Determined is the user doesn't need to keep track of having to average work.

Reply:
evaluate_batch still calls the full validation dataset. Is there a drastic time difference between using the evaluate_full_dataset and the evaluate_batch?

Reply:
i learned that both evaluate_batch and evaluate_full_dataset loop through the entire validation with the difference being evaluate_full_dataset gives more control at the cost of losing parallelism